### PR TITLE
DSND-2231: Apply SonarQube Reporting Fix

### PR DESCRIPTION
**Pull request by sthompsonCH**
* Updated pom to simply jacoco reporting by only using one exec file for both itests and unit tests.
* Updated Makefile so that, when the analyse-pull-request job is run on concourse, it runs both unit tests and itests and produces comprehensive test covarge in jacoco for SonarQube to analyse.
* Documented this in the README.

[DSND-2131](https://companieshouse.atlassian.net/browse/DSND-2231)

[DSND-2131]: https://companieshouse.atlassian.net/browse/DSND-2131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -288,28 +287,29 @@
...
 				<executions>
 					<execution>
 						<id>pre-unit-test</id>
 						<goals>
 							<goal>prepare-agent</goal>
 						</goals>
 						<configuration>
-							<!-- Sets the path to the file which contains the execution model. -->
-							<destFile>${sonar.jacoco.reports}/jacoco.exec</destFile>
 							<propertyName>surefireArgLine</propertyName>
```
Comment (johncook-ch) [inline pom.xml:304]: In other POMs this uses placeholder syntax like
```
<propertyName>surefireArgLine</propertyName>
```
Can you check which is correct?

File: pom.xml [hunk 2]
```diff
@@ -318,23 +318,15 @@
 							<goal>prepare-agent</goal>
 						</goals>
 						<configuration>
-							<!-- Sets the path to the file which contains the execution data. -->
-							<destFile>${sonar.jacoco.reports}/jacoco-it.exec</destFile>
 							<propertyName>failsafeArgLine</propertyName>
```
Comment (johncook-ch) [inline pom.xml:321]: In other POMs this uses placeholder syntax like
```
<propertyName>failsafeArgLine</propertyName>
```
Can you check which is correct?

**Review by johncook-ch — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (sthompsonCH) [reply pom.xml:321]: This seems to be consistent across other poms

Comment (sthompsonCH) [reply pom.xml:321]: metrics-consumer uses these property names also

**Review by cgregoryCH — APPROVED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Review by johncook-ch — APPROVED** [review]

**Pull request merged**